src/mlpack/core/cv/metrics/top_k_accuracy_impl.hpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file top_k_accuracy_impl.hpp
+ * @author Arunav Shandeelya
+ * 
+ * The implementation of class Top_K_Accuracy_score
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_CORE_CV_METRICS_TOPK_ACCURACY_IMPL_HPP
+#define MLPACK_CORE_CV_METRICS_TOPK_ACCURACY_IMPL_HPP
+
+namespace mlpack {
+namespace cv {
+
+template<typename MLAlgorithm, typename DataType, typename TopK>
+double TopKAccuracy::Evaluate(MLAlgorithm& model,
+                        const DataType& data,
+                        const arma::Row<size_t> labels,
+                        const TopK& k)
+{
+  if (data.n_cols != labels.n_cols)
+  {
+    std::ostringstream fpp;
+    fpp << "TopK_Accuracy::Evaluate(): number of points (" << data.n_cols << ")"
+        << "doesn't matches number of labels (" << labels.n_cols << ")!"
+        << std::endl;
+    throw std::invalid_argument(fpp.str());
+  }
+  arma::Row<size_t> predictedLabels;
+  // Taking Classification output from the model.
+  model.Classify(data, predictedLabels);
```

Comment:
Maybe I missed something but  `Classify` should return one label per data point and not multiple labels, so I guess you might want to get the probability for each class instead and take the score of the the first k?

Reply:
@zoq Yeah, you are right all I want is to return the maximum probability score of k classes. I got your point. Is this method is incorrect?

Reply:
@Shandilya21 I think you intended `predictedLabels` to have multiple labels, but actually the method `Classify()` of `NaiveBayesClassifier` class that you used in your test returns a label that has a maximum of log-likelihood. That means Top-1 accuracy.
https://github.com/mlpack/mlpack/blob/9672d5d15368aca5600e6f72cb8f2fd15d2bc25d/src/mlpack/methods/naive_bayes/naive_bayes_classifier_impl.hpp#L284-L305

 For Top-k accuracy, I think you might want to use `Classify()` that returns prediction probabilities. 
https://github.com/mlpack/mlpack/blob/9672d5d15368aca5600e6f72cb8f2fd15d2bc25d/src/mlpack/methods/naive_bayes/naive_bayes_classifier_impl.hpp#L307-L348

Reply:
Thanks for your comment,@KimSangYeon-DGU Well Yeah, I got your point. I thought that method I was using to `Classify` will return prediction probabilities of each class such that I can take the Top K score. Even, I need to change the` top_k_accuracy_impl.hpp` as well?

Reply:
> Even, I need to change the top_k_accuracy_impl.hpp as well?
Yes, I think it needs to be changed to get  prediction probability matrix and it should be sorted per data. After sorting, the next step is to check if the ground truth is included among the top `k`th labels.

Reply:
@KimSangYeon-DGU Yeah, I got your point. I just need to change some of the lines and will commit the code. Thanks

Reply:
@KimSangYeon-DGU Is the latest commit seems good or do I need to change it as `model. Probabilities` to return the probabilities?

Reply:
@Shandilya21 Thanks for the work. I think it's better to use `model.Probabilities()` as @zoq said. By doing that, we don't need to pass `const arma::Row<size_t> predsprob` in `TopKAccuracy::Evaluate()`, because when we initialize a classifier, it's trained in its constructor.

Reply:
@Shandilya21 Thanks for the work. Actually, I didn't intend the change you've done and it will still not work. I intended to use a `Classify()` method with data, predicted label, and prediction probability to get the predicted result and probability matrix.